src/wallet/wallet.cpp

```diff
@@ -2497,178 +2403,201 @@ CAmount CWalletObjBase::GetDebit(const isminefilter& filter) const
     return debit;
 }
 
-#if 0
-CAmount CWalletTx::GetCredit(const isminefilter& filter) const
-#else
-CAmount CWalletObjBase::GetCredit(const isminefilter& filter) const
-#endif
+bool CWalletObjBase::HasMatureOutputs() const
 {
-    // Must wait until coinbase is safely deep enough in the chain before valuing it
-    if (IsCoinBase() && !IsMature())
-        return 0;
-
-    int64_t credit = 0;
-    if (filter & ISMINE_SPENDABLE)
+    for(unsigned int pos = 0; pos < GetVout().size(); ++pos)
     {
-        // GetBalance can assume transactions in mapWallet won't change
-        if (fCreditCached)
-            credit += nCreditCached;
-        else
-        {
-            nCreditCached = pwallet->GetCredit(*this, ISMINE_SPENDABLE);
-            fCreditCached = true;
-            credit += nCreditCached;
+        switch(this->IsOutputMature(pos)) {
+        case COutputEntry::maturityState::MATURE:
+            return true;
+
+        case COutputEntry::maturityState::IMMATURE:
+            continue;
+
+        case COutputEntry::maturityState::NOT_APPLICABLE:
+        default:
+            return false;
```

Comment:
Why should maybe continue in this case, instead of returning method wth false value

Reply:
`COutputEntry::maturityState::NOT_APPLICABLE` applies to coins not in mempool nor in mainchain. As such it is valid for all outputs